src/lib/spinel/radio_spinel.cpp

```diff
@@ -1919,6 +1919,7 @@ void RadioSpinel::CalcRcpTimeOffset(void)
      *         D = T1' - ((T0 + T2)/ 2)
      */
 
+    EXPECT((mState != kStateDisabled) && (mState != kStateSleep), NO_ACTION);
```

Comment:
Do we want to skip this when radio is in  `kStateSleep`?

Reply:
The radio is enabled by default, the `mState` won't be set to `kStateDisabled `, unless users call the command `radio disable`. The condition `mState != kStateDisabled` is always `ture`.  So we can't only use the condition `mState != kStateDisabled` here. 

The Current Thread host on POSIX doesn't support the CSL receiver. The Thread host won't enter the sleep mode unless the thread network interface is down. So I'm using these two conditions to stop the time sync.

Reply:
How about calling `otThreadGetDeviceRole()` here to check the current role?

Reply:
Or maybe the `otIp6IsEnabled()`? (which tell us if interfance is up/down)?

Reply:
How about calling the `otSetStateChangedCallback()` to register a state changed callback function in `src/posix/platform/radio.cpp`. When the Thread interface state is changed, calling the new API `RadioSpinel::SetTimeSyncEnable()` to enable or disable the time sync based on the return value of `otIp6IsEnabled()`.

Reply:
Yes. This can work and would be better.

One thing to be careful is number of `OPENTHREAD_CONFIG_MAX_STATECHANGE_HANDLERS` and that we have  one reserved for the `RadioSpinel` use.

Reply:
I have updated the code to enable/disable the time sync via the function `otSetStateChangedCallback()` .